### ISSUE TITLE
Multi-class segmentation setting

### ISSUE THREAD

**Issue by juliadietlmeier** [SJTU-Intelligent-Optics-Lab/Annotation-efficient-learning-for-OCT-segmentation#3]
Is it possible to use the model in a multi-class segmentation setting? Do we only need to change the argument “out_channels=number_classes” or something else needs to be changed like loss function, metrics, softmax? In which format should be the labels? Should the multi-class labels be one-hot encoded or rather class-indexed (class1=pixel value 1, class2=pixel value 2 and so on)? Thank you!

**Comment by SJTU-Intelligent-Optics-Lab** [comment]
Hi, the model is also can be used for multi-class segmentationm, sorry in this code we didn't make use of multi-class task directly. It's not hard to change it. You can modify something as you mentioned. The loss function can be changed from “BCEWithLogitsLoss() ” to “CrossEntropyLoss()”. Moreover, the output of the current decoder is only for one class, you will first set the argument “out_channels=number_classes” as you want, and you don't have to load the pretrained weight of last output layer of decoder, and then fine-tune the model on your own dataset. For the format of multi-labels, you can do it as you want and it depends on personal habits. One-hot format is a popular way and recommend it for label-encoding. Thanks for your interesting.
